include/rocksdb/public_port.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+#pragma once
+
+// Best used after all the headers in case windows #defines give you trouble
+// Minimum subset
+
+#ifndef ROCKSDB_PUBLIC_PORT_H__
+#define ROCKSDB_PUBLIC_PORT_H__
+
+#ifdef OS_WIN
```

Comment:
So we have on choice then is Win32 and/or _MSVER which as predefined by the compiler?

Dmitri

From: Igor Canadi [mailto:notifications@github.com]
Sent: Monday, July 13, 2015 9:43 PM
To: facebook/rocksdb
Cc: Dmitri Smirnov
Subject: Re: [rocksdb] Ensure Windows build w/o port/port.h in public headers (#652)


In include/rocksdb/public_port.h<https://github.com/facebook/rocksdb/pull/652#discussion_r34536787>:

> @@ -0,0 +1,42 @@

> +// Copyright (c) 2011 The LevelDB Authors. All rights reserved.

> +// Use of this source code is governed by a BSD-style license that can be

> +// found in the LICENSE file. See the AUTHORS file for names of contributors.

> +#pragma once

> +

> +// Best used after all the headers in case windows #defines give you trouble

> +// Minimum subset

> +

> +#ifndef ROCKSDB_PUBLIC_PORT_H__

> +#define ROCKSDB_PUBLIC_PORT_H__

> +

> +#ifdef OS_WIN

This will not work in public header files since public header files are not compiled. We don't pass the defines to our customers.

—
Reply to this email directly or view it on GitHub<https://github.com/facebook/rocksdb/pull/652/files#r34536787>.